pkg/query/internal/test-storeset-pre-v0.8.0/storeset_test.go

```diff
@@ -186,7 +186,7 @@ func TestPre0_8_0_StoreSet_AgainstNewStoreGW(t *testing.T) {
 	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
 	logger = level.NewFilter(logger, level.AllowDebug())
 	logger = log.With(logger, "ts", log.DefaultTimestampUTC, "caller", log.DefaultCaller)
-	storeSet := NewStoreSet(logger, nil, specsFromAddrFunc(st.StoreAddresses()), testGRPCOpts, time.Minute)
+	storeSet := NewStoreSet(logger, nil, specsFromAddrFunc(st.StoreAddresses(), IsAgentEnabled), testGRPCOpts, time.Minute)
```

Comment:
As the error suggested, the variable `IsAgentEnabled` is not defined. Sidecar needs to automatically detect it based on the result of the flags API.

Reply:
Thank you @yeya24 I think I headed to find the solution for this issue in wrong direction sorry for the inconvenience I will Close this PR and open a new PR and work on this issue to fix it as soon as possible.

Reply:
Thank you @yeya24 I think I headed to find the solution for this issue in wrong direction sorry for the inconvenience I will work on this to fix it as soon as possible.